src/rgw/rgw_rest.cc

```diff
@@ -657,6 +657,10 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
 
   dump_trans_id(s);
 
+  if(s->prot_flags & RGW_REST_SWIFT) {
+    dump_header(s, "X-Openstack-Request-Id", s->trans_id);
+  }
+
```

Comment:
i would put this part inside `dump_trans_id()`, where there's already a section for `RGW_REST_SWIFT`